server/src/main/java/org/opensearch/action/admin/cluster/shards/routing/weighted/put/TransportAddWeightedRoutingAction.java

```diff
@@ -0,0 +1,128 @@
+    private volatile List<String> awarenessAttributes;
+
+    @Inject
+    public TransportAddWeightedRoutingAction(
+        Settings settings,
+        ClusterSettings clusterSettings,
+        TransportService transportService,
+        ClusterService clusterService,
+        WeightedRoutingService weightedRoutingService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver
+    ) {
+        super(
+            ClusterAddWeightedRoutingAction.NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            ClusterPutWeightedRoutingRequest::new,
+            indexNameExpressionResolver
+        );
+        this.weightedRoutingService = weightedRoutingService;
+        this.awarenessAttributes = AwarenessAllocationDecider.CLUSTER_ROUTING_ALLOCATION_AWARENESS_ATTRIBUTE_SETTING.get(settings);
+        clusterSettings.addSettingsUpdateConsumer(
+            AwarenessAllocationDecider.CLUSTER_ROUTING_ALLOCATION_AWARENESS_ATTRIBUTE_SETTING,
+            this::setAwarenessAttributes
+        );
+    }
+
+    List<String> getAwarenessAttributes() {
+        return awarenessAttributes;
+    }
+
+    private void setAwarenessAttributes(List<String> awarenessAttributes) {
+        this.awarenessAttributes = awarenessAttributes;
+    }
+
+    @Override
+    protected String executor() {
+        return ThreadPool.Names.SAME;
+    }
+
+    @Override
+    protected ClusterPutWeightedRoutingResponse read(StreamInput in) throws IOException {
+        return new ClusterPutWeightedRoutingResponse(in);
+    }
+
+    @Override
+    protected void clusterManagerOperation(
+        ClusterPutWeightedRoutingRequest request,
+        ClusterState state,
+        ActionListener<ClusterPutWeightedRoutingResponse> listener
+    ) throws Exception {
+        verifyAwarenessAttribute(request.getWeightedRouting().attributeName());
+        weightedRoutingService.registerWeightedRoutingMetadata(
+            request,
+            ActionListener.delegateFailure(
+                listener,
+                (delegatedListener, response) -> {
+                    delegatedListener.onResponse(new ClusterPutWeightedRoutingResponse(response.isAcknowledged()));
+                }
+            )
+        );
+    }
+
+    private void verifyAwarenessAttribute(String attributeName) {
+        // Currently, only zone is supported
+        if (!getAwarenessAttributes().contains(attributeName) || !attributeName.equalsIgnoreCase("zone")) {
+            ActionRequestValidationException validationException = null;
+
+            validationException = addValidationError(
+                String.format(Locale.ROOT, "invalid awareness attribute %s requested for updating weighted routing", attributeName),
+                validationException
+            );
+            throw validationException;
+        }
+    }
```

Comment:
I don't think we can hardcode "zone". It should be a valid configured attribute name. Also to let you know there can be multiple such values supported

Reply:
Checking just for attribute name in comma separated list of attributes